src/deluge/gui/views/automation_instrument_clip_view.h

```diff
@@ -144,12 +143,18 @@ class AutomationInstrumentClipView final : public ClipView, public InstrumentCli
 	ModelStackWithAutoParam* getModelStackWithParam(ModelStackWithTimelineCounter* modelStack, InstrumentClip* clip,
 	                                                int32_t paramID = 0xFFFFFFFF,
 	                                                Param::Kind paramKind = Param::Kind::NONE);
+	int32_t getEffectiveLength(ModelStackWithTimelineCounter* modelStack);
+	uint32_t getMiddlePosFromSquare(ModelStackWithTimelineCounter* modelStack, int32_t xDisplay);
 
+	void getParameterName(char* parameterName);
 	int32_t getParameterKnobPos(ModelStackWithAutoParam* modelStack, uint32_t pos);
 
 	bool getNodeInterpolation(ModelStackWithAutoParam* modelStack, int32_t pos, bool reversed);
 	void setParameterAutomationValue(ModelStackWithAutoParam* modelStack, int32_t knobPos, int32_t squareStart,
-	                                 int32_t xDisplay, int32_t effectiveLength, bool displayValue = true);
+	                                 int32_t xDisplay, int32_t effectiveLength);
+	void setKnobIndicatorLevels(int32_t knobPos);
+	void updateModPosition(ModelStackWithAutoParam* modelStack, uint32_t squareStart, bool updateDisplay = true,
```

Comment:
new function to update the position in the mod controllable stack so that you audition the automation at the last selected mod position